include/random.h

```diff
@@ -1,20 +1,88 @@
 #ifndef GUARD_RANDOM_H
 #define GUARD_RANDOM_H
 
-extern u32 gRngValue;
-extern u32 gRng2Value;
+// The number 1103515245 comes from the example implementation of rand and srand
+// in the ISO C standard.
+#define ISO_RANDOMIZE1(val)(1103515245 * (val) + 24691)
+#define ISO_RANDOMIZE2(val)(1103515245 * (val) + 12345)
+
+/* Notes about new functions:
+* If using HQ_RANDOM, you MUST call AdvanceRandom() in VBlank handlers.
+* If you do not, you risk corruption of the RNG state.
```

Comment:
The corruption would happen if a partially updated state is used, and it’s corruption in the sense that the generator isn’t designed to work that way and the results would be unpredictable, if difficult to notice in practice.

Reply:
AAAH! I see, I misunderstood the comment! I read it as:
> you must call `AdvanceRandom()` in the v-blank handler [because it's required to maintain an invariant of the state, which would otherwise be corrupted]

which didn't make sense, but actually it's:
> you must not call random functions _except_ `AdvanceRandom()` in the v-blank handler [because they could corrupt the state]

is that right? If so, do you think it'd be worth rewording the comment slightly?

Reply:
Yes and yes.